### DISCUSSION TITLE
CI: build Docker failing

### DISCUSSION THREAD

**Discussion by daniellepintz** [Lightning-AI/pytorch-lightning#9275] [General]
I noticed two CI jobs are consistently failing - `CI build Docker / build-Conda (3.9, 1.9) and (3.9, 1.10)`
with the error 
```
Error: buildx failed with: error: failed to solve: executor failed running [/bin/bash -c pip install --no-cache-dir --global-option="--cuda_ext" https://github.com/NVIDIA/apex/archive/refs/heads/master.zip &&     python -c "from apex import amp"]: exit code: 1
```

<img width="461" alt="Screen Shot 2021-09-01 at 9 45 18 PM" src="https://user-images.githubusercontent.com/38207072/131783955-2d2dbf71-5c4c-44d0-b17a-5dae9bd68ed0.png">

https://github.com/PyTorchLightning/pytorch-lightning/runs/3483024689
https://github.com/PyTorchLightning/pytorch-lightning/runs/3483024738

Starting this discussion to track and try to resolve. anyone have any ideas? cc @Borda

**Comment by Borda** [comment]
seems like failing to install Apex, either on the build/install or in testing whether it is correctly installed...
could you try to build it locally and check some fixing?
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/dockers/README.md
as other Conda dockers are passing I would guess that the Apex is not compatible with PT 1.9+

**Reply by carmocca** [reply]
I'm seeing lots of recent issues about installation in https://github.com/NVIDIA/apex/issues

They might've just broken master.

**Reply by Borda** [reply]
Let's freeze on a version that still works...

**Comment by daniellepintz** [comment]
Thanks! I am having trouble building it locally with the command `docker image build -t pytorch-lightning:base-cuda-py3.8-pt1.9 -f dockers/base-cuda/Dockerfile --build-arg PYTHON_VERSION=3.8 --build-arg PYTORCH_VERSION=1.9 .` same thing with 3.7/1.6

```
CMake Error at /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
Could NOT find Pytorch: Found unsuitable version "", but required is at least "1.2.0" (found )
```
full error: https://gist.github.com/daniellepintz/4591231de0e72f9530463bd4badae742

is the command I'm running correct?

**Comment by daniellepintz** [comment]
cc @carmocca @SeanNaren who also contributed to the Docker readme https://github.com/PyTorchLightning/pytorch-lightning/blob/master/dockers/README.md
